contracts/feirari/RewardsDistributorAdmin.sol

```diff
@@ -0,0 +1,100 @@
+pragma solidity ^0.8.0;
+
+import "../refs/CoreRef.sol";
+import "./IRewardsDistributorAdmin.sol";
+
+contract RewardsDistributorAdmin is IRewardsDistributorAdmin, CoreRef {
+    /// @notice rewards distributor contract
+    IRewardsDistributorAdmin public rewardsDistributorContract;
+
+    /// @param coreAddress address of core contract
+    /// @param _rewardsDistributorContract admin rewards distributor contract
+    constructor(
+        address coreAddress,
+        IRewardsDistributorAdmin _rewardsDistributorContract
+    ) CoreRef(coreAddress) {
+        rewardsDistributorContract = _rewardsDistributorContract;
+        /// @notice Governance should create new admin role for this contract
+        /// and then grant this role to all AutoRewardsDistributors so that they can call this contract
+        _setContractAdminRole(keccak256("REWARDS_DISTRIBUTOR_ADMIN_ROLE"));
+    }
+
+    /**
+      * @notice Begins transfer of admin rights. The newPendingAdmin must call `_acceptAdmin` to finalize the transfer.
+      * @dev Admin function to begin change of admin. The newPendingAdmin must call `_acceptAdmin` to finalize the transfer.
+      * @param newPendingAdmin New pending admin.
+      */
+    function _setPendingAdmin(address newPendingAdmin) external override onlyGovernor {
+        rewardsDistributorContract._setPendingAdmin(newPendingAdmin);
+    }
+
+    /**
+      * @notice Accepts transfer of admin rights. msg.sender must be pendingAdmin
+      * @dev Admin function for pending admin to accept role and update admin
+      */
+    function _acceptAdmin() external override {
+        rewardsDistributorContract._acceptAdmin();
+    }
+
+    /*** Comp Distribution ***/
+   /*** Comp Distribution Admin ***/
+
```

Comment:
Add method here that calls _setImplementation as this will hit a proxy contract.